src/swarm-engine/Swarm/Game/State/Substate.hs

```diff
@@ -428,7 +428,9 @@ initDiscovery =
     , -- This does not need to be initialized with anything,
       -- since the master list of achievements is stored in UIState
       _gameAchievements = mempty
-    , _structureRecognition = StructureRecognizer (RecognizerAutomatons mempty mempty) emptyFoundStructures []
+    , _structureRecognition =
+        StructureRecognizer (RecognizerAutomatons mempty mempty) $
+          RecognitionState emptyFoundStructures []
```

Comment:
Personally I find the `$` more confusing than helpful here.  It's much easier to see what's going on (IMHO) with more symmetric code like
```suggestion
        StructureRecognizer 
          (RecognizerAutomatons mempty mempty)
          (RecognitionState emptyFoundStructures [])
```
But this is just me nitpicking about which shade of blue to use on the bikeshed, so feel free to ignore.